server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -283,6 +326,56 @@ private String getPropValue(Map<Integer, String> propValues, Integer resourceKey
                 return val != null && !val.isBlank() ? val.trim() : defaultValue;
             }
 
+            private ContextResource getJmsResource()
+            {
+                JmsProperties jmsProps = jmsSource();
+                if (jmsProps.getBrokerURL() == null)
+                {
+                    return null;
+                }
+
+                ContextResource jmsResource = new ContextResource();
+                jmsResource.setName("jms/ConnectionFactory");
+                jmsResource.setAuth("Container");
+                jmsResource.setType(jmsProps.getType());
+                jmsResource.setProperty("factory", jmsProps.getFactory());
+                jmsResource.setProperty("description", jmsProps.getDescription());
+                jmsResource.setProperty("brokerURL", jmsProps.getBrokerURL());
+                jmsResource.setProperty("brokerName", jmsProps.getBrokerName());
+                return jmsResource;
+            }
+            private ContextResource getLdapResource()
+            {
+                LdapProperties ldapProps = ldapSource();
+                if (ldapProps.getHost() == null)
+                {
+                    return null;
+                }
+
+                ContextResource ldapResource = new ContextResource();
+                ldapResource.setName("ldap/ConfigFactory");
+                ldapResource.setAuth("Container");
+                ldapResource.setType(ldapProps.getType());
+                ldapResource.setProperty("factory", ldapProps.getFactory());
+                ldapResource.setProperty("host", ldapProps.getHost());
+                ldapResource.setProperty("port", Integer.toString(ldapProps.getPort()));
+                if (ldapProps.getPrincipal() != null)
+                {
+                    ldapResource.setProperty("principal", ldapProps.getPrincipal());
+                }
+                if (ldapProps.getCredentials() != null)
+                {
+                    ldapResource.setProperty("credentials", ldapProps.getCredentials());
+                }
+                ldapResource.setProperty("useSsl", Boolean.toString(ldapProps.isUseSsl()));
+                ldapResource.setProperty("useTls", Boolean.toString(ldapProps.isUseTls()));
+                if (ldapProps.getSslProtocol() != null)
+                {
+                    ldapResource.setProperty("sslProtocol", ldapProps.getSslProtocol());
+                }
+                return ldapResource;
+            }
```

Comment:
Thanks for taking care of this, @labkey-jeckels . I know I'm late to the party, but it seems unfortunate that `LabKeyServer` must know about every possible property used by every module. Shouldn't we introduce a registration mechanism for modules to handle the properties they care about? Similar to startup properties.

Reply:
I don't like needing centralized knowledge of these properties either, but this code has no idea about modules. And the module code doesn't know anything about application.properties or how to find it.

Some of these properties, such as disabling 2FA, could be handled via startup properties. That may be worth doing.

JMS and LDAP could probably be handled differently too but this was the approach I found that was as close to the old mechanism as possible, and easiest to manage while we're in transition between standalone and embedded.